### PR TITLE
Feat/original split + Fix/cached download

### PR DESCRIPTION
**Pull request by mauricekraus**
This adds the ability to select the original data split, and this MR also fixes some implementation errors when downloading and caching datasets.

### CODE REVIEW

File: torchchronos/download.py [hunk 1]
```diff
@@ -1,40 +1,76 @@
...
+        sub_dir = os.path.join(data_dir, name)
+        if os.path.isdir(sub_dir):
+            all_files = os.listdir(sub_dir)
+            if name + "_TRAIN.ts" in all_files and name + "_TEST.ts" in all_files:
+                datasets.append(name)
+    return datasets
+
+
+def download_uea_ucr(
+    extract_path: str | Path | None = None, dataset_name: str = "GunPoint"
```
Comment (felixdivo) [inline torchchronos/download.py]: I would not have a default here. In partical applications, it is not very useful.
```suggestion
    extract_path: str | Path | None = None, dataset_name: str
```

File: torchchronos/download.py [hunk 2]
```diff
@@ -1,40 +1,76 @@
...
+) -> bool:
+    """
+    Downloads the UEA/UCR time series dataset from sktime to the specified path.
+
+    Parameters:
+    extract_path (str | Path | None, default=None) : The path to extract the dataset. If None, extract to `.cache/data` directory.
+    dataset_name (str, default="GunPoint"): The name of the dataset to be downloaded.
```
Comment (felixdivo) [inline torchchronos/download.py]: Usually, types & defaults are inferred from the signature (e.g. for renederingt docs) and do not need to be stated again.

**Review by felixdivo — APPROVED** [review]
🥳

File: torchchronos/download.py (hunk 1 above)
Comment (felixdivo) [reply torchchronos/download.py]: (You'd have to switch the argument order too.)

Comment (mauricekraus) [reply torchchronos/download.py]: I know, but I wanted to keep the order, but yes I can do this

File: torchchronos/download.py (hunk 2 above)
Comment (mauricekraus) [reply torchchronos/download.py]: Usually is a bold statement,
this depends on which docstring standard you look at and even there it is not defnied explicitly, however chat gpt wrote it for me :D

**Pull request merged**